integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/mv/DataMapListeners.scala

```diff
@@ -139,3 +148,130 @@ object LoadPostDataMapListener extends OperationEventListener {
+object DataMapDropColumnPreListener extends OperationEventListener {
+  /**
+   * Called on a specified event occurrence
+   *
+   * @param event
+   * @param operationContext
+   */
+  override def onEvent(event: Event, operationContext: OperationContext): Unit = {
+    val dropColumnChangePreListener = event.asInstanceOf[AlterTableDropColumnPreEvent]
+    val carbonTable = dropColumnChangePreListener.carbonTable
+    val alterTableDropColumnModel = dropColumnChangePreListener.alterTableDropColumnModel
+    val columnsToBeDropped = alterTableDropColumnModel.columns
+    if (CarbonTable.hasMVDataMap(carbonTable)) {
+      val dataMapSchemaList = DataMapStoreManager.getInstance
+        .getDataMapSchemasOfTable(carbonTable).asScala
+      for (dataMapSchema <- dataMapSchemaList) {
+        if (null != dataMapSchema && !dataMapSchema.isIndexDataMap) {
+          val listOfColumns = DataMapListeners.getDataMapTableColumns(dataMapSchema, carbonTable)
+          val columnExistsInChild = listOfColumns.collectFirst {
+            case parentColumnName if columnsToBeDropped.contains(parentColumnName) =>
+              parentColumnName
+          }
+          if (columnExistsInChild.isDefined) {
+            throw new UnsupportedOperationException(
+              s"Column ${ columnExistsInChild.head } cannot be dropped because it exists " +
+              s"in mv datamap ${ dataMapSchema.getRelationIdentifier.toString }")
+          }
+        }
+      }
+    }
+    if (DataMapUtil.isMVDataMapTable(carbonTable)) {
+      throw new UnsupportedOperationException(
+        s"Cannot drop columns present in MV datamap table ${ carbonTable.getDatabaseName }." +
+        s"${ carbonTable.getTableName }")
+    }
+  }
+}
+
+object DataMapChangeDataTypeorRenameColumnPreListener
+  extends OperationEventListener {
+  /**
+   * Called on a specified event occurrence
+   *
+   * @param event
+   * @param operationContext
+   */
+  override def onEvent(event: Event, operationContext: OperationContext): Unit = {
+    val colRenameDataTypeChangePreListener = event
+      .asInstanceOf[AlterTableColRenameAndDataTypeChangePreEvent]
+    val carbonTable = colRenameDataTypeChangePreListener.carbonTable
+    val alterTableDataTypeChangeModel = colRenameDataTypeChangePreListener
+      .alterTableDataTypeChangeModel
+    val columnToBeAltered: String = alterTableDataTypeChangeModel.columnName
+    if (CarbonTable.hasMVDataMap(carbonTable)) {
+      val dataMapSchemaList = DataMapStoreManager.getInstance
+        .getDataMapSchemasOfTable(carbonTable).asScala
+      for (dataMapSchema <- dataMapSchemaList) {
+        if (null != dataMapSchema && !dataMapSchema.isIndexDataMap) {
+          val listOfColumns = DataMapListeners.getDataMapTableColumns(dataMapSchema, carbonTable)
+          if (listOfColumns.contains(columnToBeAltered)) {
+            throw new UnsupportedOperationException(
+              s"Column $columnToBeAltered exists in a MV datamap. Drop MV datamap to " +
```

Comment:
Don't hard code mv name here, put schema name and provider name in log

Reply:
done